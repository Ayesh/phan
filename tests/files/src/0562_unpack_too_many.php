<?php

class Example {
    public function __construct() {
    }
    public static function make(...$args) {
        return new self(...$args);
    }
    public static function make2(...$args) {
        return new self('extra', ...$args);
    }
    public static function make3(...$args) {
        if ($args) {
            return new self(...$args);  // infers there's at least one argument, warns
        } else {
            return new self(...$args);  // infers there's exactly 0 arguments, doesn't warn
        }
    }
}
