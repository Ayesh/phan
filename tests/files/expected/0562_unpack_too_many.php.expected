%s:7 PhanParamTooManyUnpack Call with 0 or more args to \Example::__construct() which only takes 0 arg(s) defined at %s:4 (argument unpacking was used)
%s:10 PhanParamTooMany Call with 1 arg(s) to \Example::__construct() which only takes 0 arg(s) defined at %s:4
%s:14 PhanParamTooMany Call with 1 arg(s) to \Example::__construct() which only takes 0 arg(s) defined at %s:4
